crates/core/src/token.rs

```diff
@@ -670,17 +734,17 @@ impl From<Amount> for U256 {
     }
 }
 
-impl From<Dec> for Amount {
-    fn from(dec: Dec) -> Amount {
+impl TryFrom<Dec> for Amount {
+    type Error = ();
```

Comment:
why `()` for the `Error` type?

Reply:
ah, I'll switch it to use arith err

Reply:
fixed in [70daab5](https://github.com/anoma/namada/pull/3074/commits/70daab5da619e8028203c93d7197ef9d5970b184)